### ISSUE TITLE
feature: proxy support

### ISSUE THREAD

**Issue by maciesse** [webcatalog/webcatalog-legacy#537]
Issue: when trying to install an app behind a corporate proxy the operation fail with a timeout. 

# Feature Request
Add a proxy with authentication setting in the setting panel

**Comment by Bennett-Yang** [comment]
I really need it too. Using webcatalog in China is killing me because some websites are blocked by government. I can deal with it on Windows but I have no idea how to launch it behind a proxy on Linux.

**Comment by quanglam2807** [comment]
Thanks for the feedback!! I'll work on this. Would an option to enter proxy configuration link sufficient for this feature? It would work with `http`, `https`, `sock5`, etc.

**Comment by maciesse** [comment]
It would be perfect!

**Comment by Bennett-Yang** [comment]
> Thanks for the feedback!! I'll work on this. Would an option to enter proxy configuration link sufficient for this feature? It would work with `http`, `https`, `sock5`, etc.

It would be totally enough for me.

**Comment by maciesse** [comment]
As a side note, the singlebox app lack this feature too :)

**Comment by quanglam2807** [comment]
@maciesse  Definitely, I'll port the feature there as well. ETA: March 7 - March 15.

**Comment by quanglam2807** [comment]
Please give it a try, guys! https://github.com/quanglam2807/webcatalog/releases/tag/v19.4.0

**Comment by maciesse** [comment]
Thanks for the quick intervention. 
Having the proxy set "gave birth" to a new and related issue. The corporate proxy does https inspection (this a GDPR related activity that companies in the EU are putting in place) and use a self-signed certificate. This raise the "unable to get local issuer certificate" issue.
In my node environment I had to set this two configuration entries to bypass this problem:

`
cafile = "/path/to/cacert.pem" 
strict-ssl = false
`

Do you think is possible to have such configurations here?

**Comment by quanglam2807** [comment]
Thanks for testing! When does the bug occur? I suppose it happens with other requests, not with the webviews?

**Comment by maciesse** [comment]
It happen when trying to install a new app

**Comment by quanglam2807** [comment]
@maciesse How do you configure the CA for Chrome? As Electron is built on top of Node and Chrome. I'll need to configure that for in both Electron and Node context.

Node: https://stackoverflow.com/a/31862256
Chrome: https://stackoverflow.com/questions/44658269/electron-how-to-allow-insecure-https

**Comment by quanglam2807** [comment]
Another option is to add an option to disable certificate validation but I don't think it's preferable

**Comment by maciesse** [comment]
For chrome I did import of the self-signed "certification authority" in the browser.
Same for node as described above.

Btw: today I'm working from home, so no corporate proxy. I tryed to install something with Electron engine and got this

![Selezione_195](https://user-images.githubusercontent.com/32390897/76063618-61400d00-5f88-11ea-9987-c59820c60ffc.png)

Using another engine it worked well.

**Comment by maciesse** [comment]
For chrome I did import of the self-signed "certification authority" in the browser.
Same for node as described above.

Btw: today I'm working from home, so no corporate proxy. I tryed to install something with Electron engine and got this

![Selezione_195](https://user-images.githubusercontent.com/32390897/76064286-a57fdd00-5f89-11ea-8fbe-2c564a5ddc1f.png)

Using another engine it worked well.

**Comment by quanglam2807** [comment]
@maciesse It's fixed in https://github.com/quanglam2807/webcatalog/releases/tag/v19.4.3

**Comment by quanglam2807** [comment]
@maciesse It's possible to add custom certificate on Linux thanks to Electron API: https://www.electronjs.org/docs/api/app#appimportcertificateoptions-callback-linux. But you won't be able to remove the certificate. Is it a deal breaker?

**Comment by maciesse** [comment]
No, it's ok.

**Comment by maciesse** [comment]
> For chrome I did import of the self-signed "certification authority" in the browser.
> Same for node as described above.
> 
> Btw: today I'm working from home, so no corporate proxy. I tryed to install something with Electron engine and got this
> 
> ![Selezione_195](https://user-images.githubusercontent.com/32390897/76063618-61400d00-5f88-11ea-9987-c59820c60ffc.png)
> 
> Using another engine it worked well.

With the 19.4.3 version that problem is still open, no matter if I set the proxy or not. Do you want me to open another issue for this problem only?

**Comment by quanglam2807** [comment]
@maciesse Thank you for letting me know! I'm working on a hot fix. Let's close this issue and move the discussion about custom certificate to #595.
